src/schema/v2/profile/updateProfileMutation.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  GraphQLString,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLUnionType,
+  GraphQLBoolean,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import Profile, { ProfileType } from "../profile"
+import { ResolverContext } from "types/graphql"
+
+interface UpdateProfileMutationInputProps {
+  id: string
+  handle?: string | null
+  bio?: string | null
+  fullBio?: string | null
+  website?: string | null
+  location?: string | null
+  isPrivate?: boolean | null
+  menuColorClass?: string | null
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateProfileSuccess",
+  isTypeOf: (data) => data._id,
+  fields: () => ({
+    profile: {
+      type: ProfileType,
+      resolve: (profile) => profile,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateProfileFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "UpdateProfileResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const updateProfileMutation = mutationWithClientMutationId<
+  UpdateProfileMutationInputProps,
+  any,
+  ResolverContext
+>({
+  name: "UpdateProfileMutation",
+  description: "Updates a profile.",
+  inputFields: {
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The id of the profile to update.",
+    },
+    handle: {
+      type: GraphQLString,
+      description: "Unique handle.",
+    },
+    bio: {
+      type: GraphQLString,
+      description: "Short bio (275 character max).",
+    },
+    fullBio: {
+      type: GraphQLString,
+      description: "Full bio (800 character max).",
+    },
+    website: {
+      type: GraphQLString,
+      description: "Website.",
+    },
+    location: {
+      type: GraphQLString,
+      description: "Location.",
+    },
+    isPrivate: {
+      type: GraphQLBoolean,
+      description: "Private profiles hide certain features for non admins.",
+    },
+    menuColorClass: {
```

Comment:
ooo who is this `menuColorClass`? I dont see any reference in the UI?

Reply:
I dont see anyone setting this value in my very swift and biased [search](https://github.com/search?q=org%3Aartsy+menu_color_class&type=code) maybe we can ✂️

Reply:
Removed!